### ISSUE TITLE
Bug: popper content not positioned outside panel container

### ISSUE THREAD

**Issue by jcfranco** [Esri/calcite-components#2069]
### Actual Behavior

The popper content from some components is not positioned correctly when placed inside a panel. 

**Note**: this is likely to happen outside of calcite-panel scenarios

### Expected Behavior

Popper content to be positioned outside of calcite-panel.

### Reproduction Steps or Sample

https://codepen.io/jcfranco/pen/mdRgmzB?editors=1000

Note that the above codepen is non-exhaustive and only shows this issue w/ calcite-dropdown and calcite-tooltip

### Relevant Info

_Version_: `@esri/calcite-components@next.177`

- [x] CDN
- [ ] NPM package

Setting `strategy: "fixed"` [when creating poppers](https://github.com/Esri/calcite-components/blob/master/src/utils/popper.ts#L59) seems to fix the issue, but needs further testing/review.

cc @AdelheidF

**Comment by driskull** [comment]
@jcfranco do you think we should just expose the strategy setting on every component using popper? We don't always want it fixed otherwise it won't be hidden on internal scrolls like below.

![Apr-27-2021 15-33-26](https://user-images.githubusercontent.com/1231455/116320939-0827cf80-a76e-11eb-8dd3-3c8cc00267ab.gif)

Possible fixes:

1. expose `strategy` as defined in Popper and if more strategies are added we can expose them.
2. Expose boolean 'fixed' prop.

**Comment by driskull** [comment]
current:

![current](https://user-images.githubusercontent.com/1231455/116321188-72407480-a76e-11eb-9ba0-cb1632918633.gif)

**Comment by driskull** [comment]
related to #1913

**Comment by jcfranco** [comment]
@driskull I like option 2. Could we use another name to make it easier on users to know when to enable? Something like `appendToBody` from #1913 could work. WDYT?

**Comment by driskull** [comment]
>  Could we use another name to make it easier on users to know when to enable? WDYT?

Yes, but not appendToBody because it doesn't do that. What are you proposing?

I like strategy because it aligns with popper and if they add another option. We could use `position: absolute`?

**Comment by driskull** [comment]
@paulcpederson @caripizza @macandcheese @bstifle ^

Do you have any opinions on the name? Trying to see if there's a name that ties in popover context. Make it easier to associate that it is for the popping content and not the component itself.

**Comment by paulcpederson** [comment]
That is a really hard concept to express succinctly in a prop! Maybe `popup-positioning: "absolute" | "fixed"`?

**Comment by macandcheese** [comment]
Yeah this is a tough one, ha. Kind of hard to sum up what the resultant change will be, maybe something like positioning-type, overflow-positioning?

Strategy does seem a little hard to parse initially, Popup-positioning is reasonable to me.

**Comment by jcfranco** [comment]
Throwing in a few more into the ring:

- overlayPositioning
- popperPositioning // Popper's term for poppable content; good if we used 'popper' in our doc (not the lib) 
- openPositioning
- openingStrategy

Overall, I think all of the suggestions in this thread are more expressive than `strategy` by itself.

**Comment by driskull** [comment]
I'm good with `overlayPositioning` or `popperPositioning`. Overlay might be better if we want to not expose popper.

**Comment by AdelheidF** [comment]
I like `openPositioning`. Most users won't know what popper means and I don't think of overlay when opening a small dropdown.

**Comment by macandcheese** [comment]
To clarify, would this be a prop for all components that use popper? Combobox, Date, Modal (?), Popover, Tooltip, etc...

**Comment by driskull** [comment]
> To clarify, would this be a prop for all components that use popper? Combobox, Date, Modal (?), Popover, Tooltip, etc...

Yes, that's the idea.

> I like openPositioning. Most users won't know what popper means and I don't think of overlay when opening a small dropdown. Popup is not being used anywhere else, I think.

The only issue with 'open' is that some components use 'active', like dropdown.

**Comment by macandcheese** [comment]
`overlayPositioning` seems to work across multiple components, without the possible conflation with `active` mentioned above... I'm good with that.

**Comment by driskull** [comment]
I'm ok with 'openPositioning' if we rename all 'active' props to be 'open' :) 

It creates more work but I think we aren't really consistent with open vs active on some of these.

**Comment by macandcheese** [comment]
For all popper / "overlay" components? That's a fine distinction to me... But we use it places like `calcite-block` too where it's never an overlay. I think we probably also need to refactor Alert to use popper now that I think of it.

**Comment by driskull** [comment]
> But we use it places like calcite-block too where it's never an overlay. I think we probably also need to refactor Alert to use popper now that I think of it.

I don't think alert or block need to use popper.

**Comment by driskull** [comment]
Alert is always at the bottom and not tied to a reference element. It doesn't need any popper logic.

**Comment by jcfranco** [comment]
FWIW, block's design didn't include any overlay.

**Comment by macandcheese** [comment]
Agree with block, was an example of a non-floaty that had 'open' nomenclature. 

Alert actually has some other positioning use-cases where Popper could help, I mentioned it because it could use "absolute" in addition to fixed, as that's often requested when folks want to put an alert in a modal. I'll open a separate issue for that as it's not related to the discussion here, but should use whatever the resultant feature is.

**Comment by driskull** [comment]
Closing. reopen if still an issue.
